### DISCUSSION TITLE
Pester filters when using containers

### DISCUSSION THREAD

**Discussion by kingspider** [pester/Pester#2529] [Q&A] [answered]
Currently trying to use a pester configuration and container. However when using the container the filters (tags) are ignored. 

Any advice?

**Comment by fflaten** [comment]
Hi. Can you please provide a sample test file with the necessary tags and the command used to run Pester so we can reproduce?

**Reply by kingspider** [reply]
I've thrown together some sample code of how I've tried to implement the tests. Could be some errors, sorry


Suite.Tests.ps1
```
param (
   [Parameter(Mandatory)]
   [string] $name
)

Describe "Dynamic Test" {

   Context "Test Suite" {

       It "Should Pass"  -Tag 'Pass' {
           $name | Should -be "Pass"
       }

       It "Should fail" -Tag 'Fail' {
           $name | Should -be "Fail"
       }
   }
}

```
**Run.ps1**
```
# import module before creating the object
Import-Module Pester

$container = New-PesterContainer -Path '$PSScriptRoot/*' -Data @{ name = "Pass"  }

# get default from static property
$configuration = [PesterConfiguration]::Default

$configuration.Run.Path = '$PSScriptRoot/*'
$configuration.Run.Container = $container

$configuration.Filter.Tag = 'Pass'
$configuration.Filter.ExcludeTag = 'Fail'

Invoke-Pester -Configuration $configuration
```

**Reply by fflaten** [reply]
You need to replace the single quotes with double quotes in both your Path values so `$PSScriptRoot` can be resolved. Except for that, this works on my machine.

```powershell
❯ .\Run.ps1

Starting discovery in 1 files.
Discovery found 2 tests in 153ms.
Running tests.
[+] C:\Users\Frode\Desktop\Demo\Suite\Suite.Tests.ps1 895ms (128ms|629ms)
Tests completed in 913ms
Tests Passed: 1, Failed: 0, Skipped: 0, Inconclusive: 0, NotRun: 1
```
Only one of the tests was executed. If you add `$configuration.Output.Verbosity = 'Detailed'` you'll see that it was only "Should Pass" that was executed

What error/result do you get that indicates a problem?

**Reply by kingspider** [reply, marked as answer]
Thanks for verifying. 

🤦‍♂️after I reviewed my code and I had mistakenly removed a character from the variable name of filters applied.

**Reply by fflaten** [reply]
Glad it got resolved. Thanks for using Pester!
